### DISCUSSION TITLE
OWUI integration with Groq inference engine (not Elon Musk's Grok) - 500 tokens/sec

### DISCUSSION THREAD

**Discussion by tbendien** [open-webui/open-webui#974] [Ideas]
@tjbck @justinh-rahb have y'all seen this?

Is there a way to integrate Groq API into OWUI?
https://wow.groq.com/

**Comment by justinh-rahb** [comment]
Yes. LiteLLM:

models `groq/llama2-70b-4096` and `groq/mixtral-8x7b-32768`

**Reply by HydeAndGeek** [reply]
excuse me Justin. Kind sir. Would you be a gent and explain this in layman's terms? I have spun up docker, running ollama, i am on intel gpu, i7, 8gb ram, not doing too well here,things are superrrrrr slow, and i know groq is blazing fast, seeing if there's a way i can add her to owui. Thanks in advance seriously if you can help make this happen i will gladly cashapp like $5

**Reply by sutyrin** [reply]
How did it go?)

**Comment by rikhuijzer** [comment]
This is the configuration that I used to set up open-webui with llama3-70b running on Groq's servers. I've added three files:

### config.yaml

```yaml
model_list:
  - model_name: 'llama3-70b'
    litellm_params:
      model: 'groq/llama3-70b-8192'
      api_key: 'os.environ/GROQ_API_KEY'
```

### docker-compose.yaml

```yaml
version: '3.7'

services:
services:
  open-webui:
    container_name: 'open-webui'
    image: 'ghcr.io/open-webui/open-webui:main'
    volumes:
      - './open-webui:/app/backend/data:rw'
      - './config.yaml:/app/backend/data/litellm/config.yaml:ro'
    ports:
      - '3003:8080'
    environment:
      DEFAULT_MODELS: 'llama3-70b'
    env_file:
      - 'GROQ_API_KEY.env'
    extra_hosts:
      - 'host.docker.internal:host-gateway'
    restart: 'unless-stopped'
```

### GROQ_API_KEY.env

```sh
GROQ_API_KEY=<YOUR API KEY>
```

After running this with

```sh
$ docker compose up
```

I had a working web UI with llama3 running at http://127.0.0.1:3003.

If you're saving this in Git, then don't forget to add `**/*.env` and `**/open-webui/` to `.gitignore`. Also, set
```yaml
    environment:
      ENABLE_SIGNUP: 'False'
```
if you put this configuration on a public server.

**Reply by BlissiraHQ** [reply]
Amazing Thanks!

**Comment by yjrocks712** [comment]
have a look in this video the issue is easily explained in this . https://youtu.be/qOxsIlpIMt4

**Comment by gtasteve** [comment]
### Adding the Groq API 

1. GO to Settings | Connections
2. Enter https://api.groq.com/openai/v1 for the API link and add the API key you get from Groq

![image](https://github.com/open-webui/open-webui/assets/46968401/238ab4f3-96ab-4713-a3e1-eccc46e4ab2c)

**You will see all LLMs available on Groq**

![image](https://github.com/open-webui/open-webui/assets/46968401/4a44687d-90e6-4dd4-80d9-e64fbfbf959a)

**Reply by basicallynabbo** [reply]
thank you sir

**Comment by yjrocks712** [comment]
did you watch this video ? https://youtu.be/qOxsIlpIMt4
